src/main/java/uk/gov/pay/ledger/agreement/entity/AgreementsFactory.java

```diff
@@ -32,4 +45,10 @@ public PaymentInstrumentEntity createPaymentInstrument(EventDigest eventDigest)
         entity.setEventCount(eventDigest.getEventCount());
         return entity;
     }
+
+    private AgreementStatus getStatus(EventDigest eventDigest) {
+        return eventDigest.getMostRecentSalientEventType()
+                .map(EVENT_TYPE_AGREEMENT_STATE_MAP::get)
+                .orElseThrow(() -> new RuntimeException("No salient event found for event digest for agreement with external ID: [xxxx]"));
```

Comment:
Need to template in the agreement external ID - my suggestion was just an example of the exception message

Reply:
D'oh, sorry